ols/src/query_helpers/docs_summarizer.py

```diff
@@ -35,8 +43,161 @@ def summarize(
             kwargs: Additional keyword arguments for customization (model, verbose, etc.).
 
         Returns:
-            A tuple containing the summary as a string and referenced documents as a string.
+            A tuple containing  Response or String and referenced documents as a string.
+        """
+        rag_sucess = True
+        if (
+            config.ols_config.reference_content is not None
+            and config.ols_config.reference_content.product_docs_index_path is not None
+        ):
+            try:
+                summary, referenced_documents = self.get_docs_with_rag(
+                    conversation_id, query, history
+                )
+                logger.info(f"{conversation_id} Summary response: {summary.response}")
+                logger.info(
+                    f"{conversation_id} Referenced documents: {referenced_documents}"
+                )
+            except Exception as err:
+                logger.info(f"Error getting answer from reference index: {err}")
+                rag_sucess = False
+
+            if rag_sucess:
+                return summary, referenced_documents
+        summary_no_rag, referenced_documents = self.get_docs_no_rag(
+            conversation_id, query, history
+        )
+        logger.info(f"{conversation_id} Summary response: {summary_no_rag}")
+        logger.info(f"{conversation_id} Referenced documents: {referenced_documents}")
+        return summary_no_rag, referenced_documents
+
+    def get_docs_no_rag(
+        self,
+        conversation_id: str,
+        query: str,
+        history: Optional[str] = None,
+    ) -> tuple[str, str]:
+        """Summarize the given query, with no RAG , based on the provided conversation context .
```

Comment:
`Summarize the given query, with no RAG, based on the provided conversation context.`